numpy/ma/tests/test_core.py

```diff
@@ -1808,6 +1808,11 @@ def test_default_fill_value_structured(self):
         assert_equal(f1, expected)
         assert_equal(f2, expected)
 
+    def test_default_fill_value_void(self):
+        dt = np.dtype([('v', 'V7')])
```

Comment:
Because that returns `bytes` and not `np.void`, so doesn't actually check that casting inside `default_fill_value` works